File: Factorial/Java/Factorial_v4.java
/* Creado por Andrés Melgar
 * para iterando++ http://iterando.online/
 * entrada: http://iterando.online/ejercicios-resueltos/factorial-en-java/
 */
package factorial_v4;

import java.util.Scanner;

public class Factorial_v4 {

    public static long factorial(int n) {
        if (n <= 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        int n;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese número n (0<=n<=20): ");
        n = reader.nextInt();

        if (n < 0 || n > 20) {
            System.out.printf("Debe ingresar un número en el rango [0..20]");
        } else {
            System.out.printf("%d!=%d%n", n, factorial(n));
        }
    }

}

File: Factorial/Python/factorial_v4.py
#Creado por Andrés Melgar
#para iterando++ http://iterando.online/
#entrada: http://iterando.online/ejercicios-resueltos/factorial-en-python/
import numpy

def inicializar_arreglo(factorial, n):
    factorial[0] = 1
    for i in range(2, n+1):
        factorial[i] = i * factorial[i - 1]

MAX_FACTORIALES=20        
n=int(input("Ingrese número n (0<=n<=20): "))
if n < 0 or n >MAX_FACTORIALES:
    print("Debe ingresar un número en el rango [0..20]")
else:    
    factorial=numpy.arange(MAX_FACTORIALES+1)
    inicializar_arreglo(factorial, MAX_FACTORIALES)
    print("{:d}!={:d}".format(n, factorial[n]))